evalml/pipelines/components/transformers/imputers/target_imputer.py

```diff
@@ -132,9 +132,15 @@ def transform(self, X, y):
         ):
             y_t = y_t.astype(bool)
 
+        new_logical_type = (
```

Comment:
Since the target is being imputed, it seems safe to assume there will be no null values in it. It should be safe to change the data type to Integer then.